docs/recipes.rst

```diff
@@ -0,0 +1,33 @@
+.. _recipes:
+
+PEX Recipes and Notes
+=====================
+
+Gunicorn and PEX
+----------------
+
+Normally, to run a wsgi-compatible application with Gunicorn, you'd just
+point Gunicorn at your application, tell Gunicorn how to run it, and you're
+ready to go - but if your application is shipping as a PEX file, you'll have
+to bundle Gunicorn as a dependency and set Gunicorn as your entry point. Gunicorn
+can't enter a PEX file to retrieve the wsgi instance, but that doesn't prevent
+the PEX from invoking Gunicorn.
+
+This retains the benefit of zero `pip install`'s to run your service, but it
+requires a bit more setup as you must ensure Gunicorn is packaged as a dependency. 
+The following snippets assume Flask as the wsgi framework, Django setup should be 
+similar:
+
+.. code-block:: bash
+
+    $ pex flask gunicorn myapp -e gunicorn.app.wsgiapp -o ~/service.pex
```

Comment:
Sorry about this - one last edit please: `pex flask gunicorn myapp -c gunicorn -o ~/service.pex`.

I do not know gunicorn and the `-e` seemed suspiciously manual for something that is presumably gunicorn bread and butter. My discovery process:
```
$ pex gunicorn -o gunicorn.pex
$ unzip -qc gunicorn.pex .deps/gunicorn-19.9.0-py2.py3-none-any.whl/gunicorn-19.9.0.dist-info/entry_points.txt

    [console_scripts]
    gunicorn=gunicorn.app.wsgiapp:run
    gunicorn_paster=gunicorn.app.pasterapp:run

    [paste.server_runner]
    main=gunicorn.app.pasterapp:paste_server
$ unzip -qc gunicorn.pex .deps/gunicorn-19.9.0-py2.py3-none-any.whl/gunicorn/app/wsgiapp.py | tail
    """\
    The ``gunicorn`` command line runner for launching Gunicorn with
    generic WSGI applications.
    """
    from gunicorn.app.wsgiapp import WSGIApplication
    WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()


if __name__ == '__main__':
    run()
```

Reply:
Oh... I spent an embarrassing amount of time researching/asking questions regarding executing a specific module under gunicorn. And I'm glad it can be this simple. I hadn't seen a line of docs on `entry_points` until our chat here, even after digging deep into the python docs on setuputils. Now knowing about it, it's aggressively obvious in the Gunicorn setup file :face_with_head_bandage: 

This PR has been a journey of enlightenment.

So that totally works like a charm and simplifies a huge pile of assumptions I've had about the way I've been hot-wiring Gunicorn. Update commit inbound.